bin-lint/shellcheck

```diff
@@ -6,9 +6,11 @@ set -euo pipefail # exit on any error, don't allow undefined variables, pipes do
 
 shebang_files=$(
   changed-not-deleted-files | \
-    rg -v '.md$' | \
-    xargs -r rg --files-with-matches '^#!/usr/bin/env (bash|sh)' || \
-    true
+    while IFS= read -r file ; do
+      if [ -f "$file" ] && head -n 1 "$file" | grep -q '^#!/usr/bin/env \(bash\|sh\)' ; then
```

Comment:
This checks only the first line for a shebang. Otherwise, shellcheck will try to lint `bin/cheet` (even though it has a zsh shebang, which shellcheck doesn't support) because it has `#!/usr/bin/env bash` on line 38.

Reply:
Because of this change, we no longer need to explicitly exclude markdown files (which, similarly, might have had a bash or sh shebang somewhere within them, but not on the first line).